lib/cpp/Kconfig

```diff
@@ -96,6 +97,15 @@ config GLIBCXX_LIBCPP
 	  Build with GNU C++ Standard Library (libstdc++) provided by the GNU
 	  Compiler Collection (GCC)-based toolchain.
 
+config LLVM_LIBCXX
+	bool "LLVM C++ Standard Library"
+	depends on !NATIVE_APPLICATION
```

Comment:
this can't be the only dependency required.

It doesn't seem right that i'm able to select LLVM C++ library when compiling with GNU ARM Embedded or ARC MWDT.

Reply:
Added dependency on Newlib and LLVM toolchain.